### PR TITLE
Add new food form

### LINKED ISSUE #4: [2.5pt] Add the form to create a new food - Team Member 1
- The application should display the form: **Create a new food** to the logged-in user, in which he can save a new food in the database. This form must have the following fields:
        1. name (required)
        2. measurement unit (required)
        3. price (required)
        4. quantity (required)

- The app should display an `Add Food` button on the **Foods List** page that allows the user to open the **Create a new food** page

- Add authorization rules if necessary

- Customize the layout by adding CSS styles

### PR DESCRIPTION
**Pull request by NitBravoA92**
# 🚩New Implementations

## Hi there! @Luffytaro22 👋

**Here I detail a summary of the new features that have been implemented in this branch:**

- [x] Update the association in the food model
- [x] Create the `food_by_user` method that allows to access the current_user's foods
- [x] Create the `new` and `create` actions in the `FoodsController` to save a new food in the database
- [x] Create the `new.html.erb` file in the foods folder that displays a form to add a new food
- [x] Create the `Add Food` button in the `index.html.erb` file in the foods folder
- [x] Update the CSS styles of the app

## ⭐To the code reviewer 👨‍💻

Please, I kindly ask you to review the entire code and verify that all requirements have been met correctly.

if there is any issue 🦯 in this PR, please do list 📃 them in a descriptive 💡 manner and give your best suggestions 🎁 if needed.

### CODE REVIEW

File: app/views/foods/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        <%= f.text_field :name, required: true, class: 'form-field' %>
+      </div>
+      <div class="form-fields-group">
+        <%= f.label 'Measurement unit' %>
+        <%= f.select :measurement_unit, options_for_select(['grams', 'milliliters', 'units']), { prompt: 'Select a  measurement' }, required: true, class: 'form-field' %>
+      </div>
+      <div class="form-fields-group">
+        <%= f.label 'Unit price' %>
+        <%= f.number_field :price, required: true, class: 'form-field' %>
+      </div>
```
Comment (Luffytaro22) [inline app/views/foods/new.html.erb]: I can't introduce the price as a decimal, only as an integer. When I try, this message pops up:
sdf

**Review by Luffytaro22 — CHANGES_REQUESTED** [review]
> ## Hi @NitBravoA92 
> Good job so far!
> There are some issues that you still need to work on to go to the next project but you are almost there!

> ## Highlights
> ✅ The food is created successfully.
> ✅ Nice-looking form.

> ## Required changes ♻️ 
> _Check the comments under the review._

File: app/views/foods/new.html.erb (hunk 1 above)
Comment (NitBravoA92) [reply app/views/foods/new.html.erb]: Hi, @Luffytaro22 Thank you so much for reviewing my code. You are absolutely right, I am going to add the necessary code to fix this error.

**Review by Luffytaro22 — APPROVED** [review]
> ## Hi @NitBravoA92 
> Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

**Pull request merged**